### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Estete9**
## Project requirements
n+1

**NOTE**
If you have problems with permissions in the tmp directory when you run `rails server`, refresh the browser's page and it should be fixed. This is an issue with user permissions and the tmp folder having an issue.

- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
  - Do not use the bullet gem in this exercise - instead analyze the logs as showed in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
  - In your PR description include the screenshots of your app's logs before and after the fix
  - This is the corresponding wireframe view: blog all user posts
image

Integration specs

- Use Capybara to write integration tests for each view in your project.
- Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.
- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.
- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commenter.
  - I can see the comment each commentor left.

## Screenshots
**BEFORE FIX**
image
image

**AFTER FIX**
image
image

### CODE REVIEW

File: spec/features/users_show_navigation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      visit user_path(@user1.id)
+    end
+
+    scenario "When I click a user's post, it redirects me to that post's show page" do
+      post_element = find('a', text: 'sixth')
+      post_element.click
+
+      expect(page).to have_current_path(user_post_path(@post6.author_id, @post6.id))
+    end
+  end
```
Comment (rloterh) [inline spec/features/users_show_navigation_spec.rb]: - [x] The test to assert the scenario, **_When I click to see all posts, it redirects me to the user's post's index page_** for _user show_ view integration spec is missing! Please ensure to improve upon this aspect.

File: spec/features/posts_show_comments_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      end
+    end
+
+    scenario 'I can see the comment each commentor left' do
+      [@comment1, @comment2].each do |comment|
+        expect(page).to have_content(comment.text.to_s)
+      end
+    end
+  end
+end
```
Comment (rloterh) [inline spec/features/posts_show_comments_spec.rb:35]: - [x] The _post show_ view integration spec fails to test for the following scenarios, let's ensure to add them accordingly:
  * I can see the post's title.
  * I can see who wrote the post.
  * I can see how many comments it has.
  * I can see how many likes it has.
  * I can see the post body.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Esteban, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🏁 🚦  ⛪ 🚜  🏃‍♀️  🌴  🏫 🌳 🏘️ 

## Highlights 👍  
- N + 1 problem addressed!
- Added integration tests are passing!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Let's make the ReadMe stand out even better by including appropriate instructions under the _Getting Started_ section, on how to run app test cases, for instance, using the command `rspec` and so on.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/posts_show_comments_spec.rb (hunk 2 above)
Comment (Estete9) [reply spec/features/posts_show_comments_spec.rb:35]: Thank you for your feedback, I created a new file for these tests `posts_show_display_spec.rb.`

**Review by julie-ify — APPROVED** [review]
Hi @Estete9 and @joramot,

## Highlights!
- Good code organization and passing specs ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**